code/atom.dm

```diff
@@ -963,6 +1006,9 @@
 //reason for having this proc is explained below
 /atom/proc/set_density(var/newdensity)
 	src.density = HAS_ATOM_PROPERTY(src, PROP_ATOM_NEVER_DENSE) ? 0 : newdensity
+	if(src.density != newdensity)
```

Comment:
![image](https://user-images.githubusercontent.com/3855802/204512401-f2f8dd60-9eea-4841-a554-8efb69e9231d.png)

Reply:
one more for the linter eh?

Reply:
Opened #12217.